tree/dataframe/src/RNTupleDS.cxx

```diff
@@ -277,7 +277,11 @@ void RNTupleDS::AddField(const RNTupleDescriptor &desc, std::string_view colName
          auto cardinalityField = std::make_unique<ROOT::Experimental::Internal::RRDFCardinalityField>();
          cardinalityField->SetOnDiskId(fieldId);
          fColumnNames.emplace_back("R_rdf_sizeof_" + std::string(colName));
-         fColumnTypes.emplace_back(cardinalityField->GetTypeName());
+         const auto typeName = cardinalityField->GetTypeName();
+         fColumnTypes.emplace_back(typeName);
+         std::string normalized;
+         TClassEdit::GetNormalizedName(normalized, typeName);
+         fNormalizedColumnTypes.emplace_back(normalized);
```

Comment:
This and further down: I wonder if instead of adding the normalized type name everywhere where we append to fColumnTypes, perhaps we should loop once over all the column types when they are fully build, i.e. at the end of the constructor.